anndata/_config.py

```diff
@@ -0,0 +1,332 @@
+from __future__ import annotations
+
+import os
+import warnings
+from collections.abc import Iterable
+from contextlib import contextmanager
+from functools import wraps
+from inspect import Parameter, signature
+from typing import TYPE_CHECKING, Any, NamedTuple, TypeVar
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Sequence
+
+T = TypeVar("T")
+
+
+class DeprecatedOption(NamedTuple):
+    option: str
+    message: str | None
+    removal_version: str | None
+
+
+# TODO: inherit from Generic[T] as well after python 3.9 is no longer supported
+class RegisteredOption(NamedTuple):
+    option: str
+    default_value: T
+    doc: str
+    validator: Callable[[T], None] | None
+
+
+def update_override_function(register: Callable):
+    """This function updates the keyword arguments, docstring, and annotations of the `SettingsManager.override` function as the `SettingsManager.register` method is called.
+
+    Parameters
+    ----------
+    register
+        This is a reference to the register function of `SettingsManager`
+
+    Returns
+    -------
+    Callable
+        The wrapped `SettingsManager.register` function.
```

Comment:
Please put types into the annotations, not into docstrings.

```suggestion
    The wrapped `SettingsManager.register` function.
```

Reply:
Missed that one and a few others when cleaning up.  Will fix them all.